src/traces/surface/defaults.js

```diff
@@ -85,7 +87,20 @@ module.exports = function supplyDefaults(traceIn, traceOut, defaultColor, layout
         }
     }
 
+    // backward compatibility block
+    if(!surfaceColor) {
+        mapLegacy(traceIn, 'zmin', 'cmin');
+        mapLegacy(traceIn, 'zmax', 'cmax');
+        mapLegacy(traceIn, 'zauto', 'cauto');
+    }
+
     colorscaleDefaults(
-        traceIn, traceOut, layout, coerce, {prefix: '', cLetter: 'z'}
+        traceIn, traceOut, layout, coerce, {prefix: '', cLetter: 'c'}
     );
 };
+
+function mapLegacy(traceIn, oldAttr, newAttr) {
+    if(oldAttr in traceIn && !(newAttr in traceIn)) {
+        traceIn[newAttr] = traceIn[oldAttr];
```

Comment:
> but I imagine that may cause problems if there are other components that make use of it somewhere still.

Exactly. That's why I took the conservative route here. We'll clean this up in `v2.0.0` :tada: .